bigquery/src/main/scala/com/yahoo/maha/executor/bigquery/BigqueryQueryExecutor.scala

```diff
@@ -0,0 +1,227 @@
+          builder
+            .setCredentials(ServiceAccountCredentials.fromStream(fileInputStream, proxyTransport))
+            .setTransportOptions(
+              HttpTransportOptions
+                .newBuilder()
+                .setHttpTransportFactory(proxyTransport)
+                .build()
+            )
+        } else {
+          builder.setCredentials(ServiceAccountCredentials.fromStream(fileInputStream))
+        }
+      } finally {
+        fileInputStream.close()
+      }
+    } else {
+      builder
+    }
+
+    builderWithCredentials
+      .build()
+      .getService
+  }
+
+  private[this] def googleApiHttpTransportFactory(): HttpTransportFactory = {
+    info(s"Setting up proxied Google API Http Transport")
+    val fileInputStream = new FileInputStream(proxyCredentialsFilePath.get)
+    try {
+      val proxyCredentials = mapper.readValue[ProxyCredentials](fileInputStream)
+      val proxyHostDetails = new HttpHost(proxyHost.get, proxyPort.get.toInt)
+      val proxyRoutePlanner = new DefaultProxyRoutePlanner(proxyHostDetails);
+      val credentialsProvider = new BasicCredentialsProvider
+      credentialsProvider.setCredentials(
+        new AuthScope(proxyHostDetails.getHostName, proxyHostDetails.getPort),
+        new UsernamePasswordCredentials(
+          proxyCredentials.userName,
+          proxyCredentials.password
+        )
+      )
+      val mHttpClient = ApacheHttpTransport.newDefaultHttpClientBuilder
+        .setRoutePlanner(proxyRoutePlanner)
+        .setRetryHandler(new DefaultHttpRequestRetryHandler)
+        .setProxyAuthenticationStrategy(ProxyAuthenticationStrategy.INSTANCE)
+        .setDefaultCredentialsProvider(credentialsProvider)
+        .build()
+
+      val mHttpTransport = new ApacheHttpTransport(mHttpClient)
+      () => mHttpTransport
+    } finally {
+      fileInputStream.close()
+    }
+  }
+}
+
+class BigqueryQueryExecutor(
+  config: BigqueryQueryExecutorConfig,
+  lifecycleListener: ExecutionLifecycleListener
+) extends QueryExecutor
+  with Logging {
+
+  val engine: Engine = BigqueryEngine
+  val bigqueryClient = config.buildBigqueryClient()
+
+  def execute[T <: RowList](
+    query: Query,
+    rowList: T,
+    queryAttributes: QueryAttributes
+  ): QueryResult[T] = {
+    val acquiredQueryAttributes = lifecycleListener.acquired(query, queryAttributes)
+    val requestModel = query.queryContext.requestModel
+    val debugEnabled = requestModel.isDebugEnabled
+
+    if (!acceptEngine(query.engine)) {
+      throw new UnsupportedOperationException(
+        s"BigqueryQueryExecutor does not support query with engine=${query.engine}"
+      )
+    }
+
+    val formattedQuery = SqlFormatter.format(
+      query.asString,
+      FormatConfig(dialect = SqlDialect.StandardSQL)
+    )
+
+    if (debugEnabled) {
+      info(s"Running query : $formattedQuery")
+    }
+
+    val queryJobConfig = QueryJobConfiguration
+      .newBuilder(formattedQuery)
+      .setUseLegacySql(false)
```

Comment:
Could this be a config setting or is the legacy sql use deprecated ?

Reply:
The query generation code is using BigQuery Standard SQL, so executing as legacy SQL probably shouldn't be an option